public/app/plugins/datasource/loki/components/monaco-query-field/MonacoQueryField.tsx

```diff
@@ -1,12 +1,12 @@
 import { css } from '@emotion/css';
```

Comment:
My first request would be to move the button and all the extra formatting code to `LokiQueryCodeEditor`. This is a component that is used in other places, and it doesn't need to know about the formatting feature, it's a Code Editor feature.

Feel free to pass down an *optional*  `value` prop.